reactor-netty-http/src/main/java/reactor/netty/http/client/Http2AllocationStrategy.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2022 VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.client;
+
+import reactor.netty.resources.ConnectionProvider;
+
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+/**
+ * HTTP/2 {@link ConnectionProvider.AllocationStrategy}.
+ *
+ * <p>This class is based on
+ * https://github.com/reactor/reactor-pool/blob/d5cb5b72cdbcbbee8d781e06972c4da21766107f/src/main/java/reactor/pool/AllocationStrategies.java#L73
+ *
+ * @author Violeta Georgieva
+ * @since 1.0.20
+ */
+public final class Http2AllocationStrategy implements ConnectionProvider.AllocationStrategy<Http2AllocationStrategy> {
+
+	public interface Builder {
+
+		/**
+		 * Build a new {@link Http2AllocationStrategy}
+		 *
+		 * @return a new {@link Http2AllocationStrategy}
+		 */
+		Http2AllocationStrategy build();
+
+		/**
+		 * Configures the maximum number of the concurrent streams that can be opened to the remote peer.
+		 * When evaluating how many streams can be opened to the remote peer,
+		 * the minimum of this configuration and the remote peer configuration is taken.
```

Comment:
maybe indicate that `-1` is a valid value for which this statement doesn't hold?
something like:
```suggestion
		 * the minimum of this configuration and the remote peer configuration is taken (unless -1 is used).
```
the sentence right below clarifies what happens when -1 is used, on top of stating it is the default.

Reply:
fixed